### PR TITLE
Added test mocks for directory report (`open_files`), `whisper.transcribe`, `pytesseract.image_to_string`

### PR DESCRIPTION
**Pull request by benjaminLuoHC**
I left the non-mocked code commented out on top of the mocked code for reference. I'll overview each mock using examples that can be run independently in a JupyTer notebook

**file_processing.File**

```py
from unittest.mock import patch
from file_processing import Directory

def test_not_opening_files_in_directory(directory_path):
    output_path = 'temp.csv'
    with patch('file_processing.File', autospec=True) as mock_file:
        dir1 = Directory(directory_path)
        dir1.generate_report(str(output_path), open_files=False)
        for call in mock_file.mock_calls:
            args, kwargs = call[1], call[2]
            assert kwargs.get('open_file') == False, "File was opened when it should not have been"

test_not_opening_files_in_directory('tests/resources/directory_test_files/')
```

This one simulates a `File` object using `autospec=True` so it will have all the attributes that a `File` object would normally have (ex. `size`, `access_time`). It tests whether each mocked object's `open_file` parameter is false when the `Directory` `open_files` parameter is false. This was the original `test_not_opening_files_in_directory` that was failing because I had added code to `__init__.py` to simplify imports. The test is able to work after changing `patch('file_processing.file.File')` to `patch('file_processing.File')`

**whisper.transcribe**

```py
from unittest.mock import patch
from file_processing import File

def test_mock_transcription(path, transcription):
    mocked_value = {
        'text': transcription,
        'language': 'en'
    }
    with patch('whisper.transcribe', return_value=mocked_value) as mock_transcribe:
        audio_file = File(path, use_transcriber=True)
        mock_transcribe.assert_called()
        assert audio_file.metadata['text'] == transcription

test_mock_transcription(
    path='tests/resources/test_files/sample_speech.aiff',
    transcription='mocked text'
)
```

This example mocks the `transcribe` function and forces the return value to be `mocked_value`. It's also important to check `mock_transcribe.assert_called()` to ensure the mock is indeed being called

**pytesseract.image_to_string**

```py
import re
import os
from unittest.mock import patch
from file_processing import File

@patch('pytesseract.image_to_string')
def test_ocr_processing_success(mock_tesseract):
    image_path = os.path.normpath('tests\\resources\\test_files\\test_ocr_text.jpg')
    mock_tesseract.return_value = 'Test OCR'
    file = File(image_path, use_ocr=True)
    result = re.sub('[^A-Za-z0-9!? ]+', '', file.metadata['ocr_text'])

    assert result == 'Test OCR'

    _, file_extension = os.path.splitext(file.file_name)

    if file_extension != '.pdf':
        mock_tesseract.assert_called_once_with(image_path)

test_ocr_processing_success()
```

This is similar to the `whisper.transcribe` mock except with syntactical changes

### CODE REVIEW

File: logging_demo.ipynb
Comment (pgaviganHC) [inline logging_demo.ipynb:1]: Ideally runtime changes don't get committed, but this is part of what happens with notebook files.

File: tests/test_transcription_decorator.py
Comment (pgaviganHC) [inline tests/test_transcription_decorator.py:1]: These tests are failing.

Comment (benjaminLuoHC) [reply tests/test_transcription_decorator.py:1]: This push overwrote the code in the transcription decorator, I'll adjust it

https://github.com/hc-sc-ocdo-bdpd/file-processing-tools/pull/152/commits/2a2fbf4ef05ec29a22eff2ceab2997b850a35cb9

Comment (pgaviganHC) [reply tests/test_transcription_decorator.py:1]: Tests passing!

Comment (BrennanHCSC) [reply tests/test_transcription_decorator.py:1]: Passing on my machine too.

**Review by BrennanHCSC — APPROVED** [review]

**Pull request merged**